src/Xamarin.Android.Build.Tasks/Utilities/ZipArchiveEx.cs

```diff
@@ -8,8 +8,8 @@ namespace Xamarin.Android.Tasks
 	public class ZipArchiveEx : IDisposable
 	{
 
-		public static int ZipFlushSizeLimit = 50 * 1024 * 1024;
-		public static int ZipFlushFilesLimit = 50;
+		public static int ZipFlushSizeLimit = 100 * 1024 * 1024;
+		public static int ZipFlushFilesLimit = 512;
```

Comment:
hmm, that might be a good idea. 🤔

Reply:
While contemplating *that* question, why are these fields read+write?  We don't ever change them, so shouldn't they be `const`?

Reply:
The idea at the start was probably that we could alter the defaults in code. But we never did lol

Reply:
@dellis1972 [wrote](https://github.com/xamarin/xamarin-android/pull/7957/files#r1166635590):

> The idea at the start was probably that we could alter the defaults in code. But we never did lol

…but you are *now*, in e.g. `AddEntryAndFlush()`.  Which means we now have unprotected globally mutable data, which I am allergic to.

For example -- and because I cannot remember -- what happens in a "redux"/"variant" of 22f10b2edec9e40a139c04564f96d8c1b3c17526 (our *favorite* issue around `.GetRegisteredTaskObjectAssemblyLocal<T>()` now needing to include a *directory* as part of the key to *prevent* unexpected sharing of data between project builds) and we have *2 or more* projects which *just happen* to hit `<BuildApk/>` at the "same" time and "just happen" to provide *different* values for `$(_ZipFlushFilesLimit)`?

It might be "fine" but I don't like it.

I would prefer it if these fields became *instance* values set at `ZipArchiveEx` construction time:

https://github.com/xamarin/xamarin-android/blob/2940d2ecd1bc870111c515263d20ca981244947a/src/Xamarin.Android.Build.Tasks/Tasks/BuildApk.cs#L142